src/Core/Domain/CustomerService/QueryResult/CustomerServiceSummary.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\CustomerService\QueryResult;
+
+/**
+ * Carries data about customer service summary: total threads and view url
+ */
+class CustomerServiceSummary
+{
+    /**
+     * @var int
+     */
+    private $contactId;
+
+    private $totalThreads = 0;
+
+    private $viewUrl = '';
+
+    /**
+     * @param int $customerThreadId
+     */
+    public function __construct(int $customerThreadId)
```

Comment:
All the parameters of this QueryResult should be in the constructor, so that it remains immutable In your handler you can prepare the value ahead of time (since they are not always fetchable) and you create your query result once you have them all available, including the default values which probably shouldn't be handle in this object but by the handler